src/unistd.rs

```diff
@@ -1324,6 +1324,42 @@ pub fn pause() {
     unsafe { libc::pause() };
 }
 
+/// The argument to [`alarm`].
+///
+/// [`alarm`]: fn.alarm.html
+pub enum SetAlarm {
+    /// Cancel a previously set alarm.
+    Cancel,
+    /// Set an alarm with second granularity.
+    Set(libc::c_uint),
+}
+
+/// Schedule or cancel an alarm signal.
+///
+/// Return the leftover time of a previously set alarm, or 0 if no alarm was
+/// set.
+///
+/// See also [alarm](http://pubs.opengroup.org/onlinepubs/9699919799/functions/alarm.html).
+///
+/// # Example
+///
+/// ```
+/// use nix::unistd::{alarm, SetAlarm};
+///
+/// // Set an alarm for 60 seconds from now.
+/// alarm(SetAlarm::Set(60));
```

Comment:
I forgot that you need to specify the enum name when you initialize a variant of it, so this doesn't read as well as I hoped. I was originally going to suggest that instead of an enum we split `alarm` into two functions, `alarm_set(c_uint)` and `alarm_cancel()`. I'm now thinking that would be a more ergonomic API. The only downside to that is discoverability, but I think the docs will be enough there. And if a user searches the docs for "alarm", I'd expect them to find these two functions and be able to understand their usage.

@Thomasdezeeuw What do you think?

Reply:
In my original pr (https://github.com/nix-rust/nix/pull/828) I had two those function exactly, but @asomers was against that.

Reply:
I didn't realize this was a spin off of an earlier PR, sorry about that. Read through it quickly and it provides a little more context. I didn't get a chance to respond early in the last PR, but I think if we have functions that start with `alarm_` it's a nice API. So let's go ahead and switch to that type of API. @asomers Care to chime in here?

Reply:
I'm also actually thinking rather than name these functions that are split off from a single `libc` function by prefixing them within their function name, instead prefix them within a module. So you'd have `alarm::set(c_uint)` and `alarm::cancel()`. That should really help with the end-user documentation.

Reply:
Actually, I like the module idea.  It's less verbose than using enums, but still Rustier than using `alarm_set` and `alarm_cancel`.

Reply:
I've put the function into their own module and also expanded the documentation.